godot/gdfreeorion.cpp

```diff
@@ -0,0 +1,237 @@
+    }
+}
+
+void GDFreeOrion::_register_methods() {
+    // register_method("_process", &GDCppTest::_process);
+    register_method("_exit_tree", &GDFreeOrion::_exit_tree);
+    register_method("_new_single_player_game", &GDFreeOrion::_new_single_player_game);
+    register_method("_get_systems", &GDFreeOrion::_get_systems);
+    register_signal<GDFreeOrion>("ping", "message", GODOT_VARIANT_TYPE_STRING);
+    register_signal<GDFreeOrion>("auth_request", "player_name", GODOT_VARIANT_TYPE_STRING, "auth", GODOT_VARIANT_TYPE_STRING);
+    register_signal<GDFreeOrion>("empire_status", "status", GODOT_VARIANT_TYPE_INT, "about_empire_id", GODOT_VARIANT_TYPE_INT);
+    register_signal<GDFreeOrion>("start_game", "is_new_game", GODOT_VARIANT_TYPE_BOOL);
+    register_property<GDFreeOrion, godot::OptionsDB*>("optionsDB",
+        &GDFreeOrion::set_options,
+        &GDFreeOrion::get_options,
+        nullptr);
+    register_property<GDFreeOrion, GodotNetworking*>("networking",
+        &GDFreeOrion::set_networking,
+        &GDFreeOrion::get_networking,
+        nullptr);
+}
+
+GDFreeOrion::GDFreeOrion() {
+}
+
+GDFreeOrion::~GDFreeOrion() {
+    // add your cleanup here
+}
+
+void GDFreeOrion::_init() {
+    InitDirs("");
+
+    GetOptionsDB().SetFromFile(GetConfigPath(), FreeOrionVersionString());
+    GetOptionsDB().SetFromFile(GetPersistentConfigPath());
+
+    CompleteXDGMigration();
+
+    // Handle the case where the resource.path does not exist anymore
+    // gracefully by resetting it to the standard path into the
+    // application bundle.  This may happen if a previous installed
+    // version of FreeOrion was residing in a different directory.
+    if (!boost::filesystem::exists(GetResourceDir()) ||
+        !boost::filesystem::exists(GetResourceDir() / "credits.xml") ||
+        !boost::filesystem::exists(GetResourceDir() / "data" / "art" / "misc" / "missing.png"))
+    {
+        DebugLogger() << "Resources directory from config.xml missing or does not contain expected files. Resetting to default.";
+
+        // GetOptionsDB().Set<std::string>("resource.path", "");
+        GetOptionsDB().Set<std::string>("resource.path", "../../default"); // Temporary default for Godot client prototype development
+
+        // double-check that resetting actually fixed things...
+        if (!boost::filesystem::exists(GetResourceDir()) ||
+            !boost::filesystem::exists(GetResourceDir() / "credits.xml") ||
+            !boost::filesystem::exists(GetResourceDir() / "data" / "art" / "misc" / "missing.png"))
+        {
+            DebugLogger() << "Default Resources directory missing or does not contain expected files. Cannot start game.";
+            throw std::runtime_error("Unable to load game resources at default location: " +
+                                     PathToString(GetResourceDir()) + " : Install may be broken.");
+        }
+    }
+
+    // initialize any variables here
+    app = std::make_unique<GodotClientApp>();
```

Comment:
@o01eg, I've moved that line back here. You were right after all, the resource folder needs to be correctly set first, because it's needed when initializing `GodotClientApp`. When the "default" folder can't be found and needs to be reset (which will happen with my hack to set it to "../../default", that happens to late and all the stuff called by the constructor of `GodotClientApp` fails.

Having no error message output in the Godot client app when the resource dir can't be found is the lesser evil in that case.

Reply:
> I've moved that line back here. You were right after all, the resource folder needs to be correctly set first, because it's needed when initializing GodotClientApp.

Yes, it's a same other clients do.